src/server/game/Server/WorldSocket.cpp

```diff
@@ -358,6 +358,8 @@ WorldSocket::ReadDataHandlerResult WorldSocket::ReadDataHandler()
         }
         case CMSG_KEEP_ALIVE:
             LogOpcodeText(opcode, sessionGuard);
+            if (_worldSession)
```

Comment:
its the session pointer in WorldSocket that needs to be protected, not any member of WorldSession